bids/layout/layout.py

```diff
@@ -299,12 +299,30 @@ def __getattr__(self, key):
 
     def __repr__(self):
         """Provide a tidy summary of key properties."""
-        # TODO: Replace each nested list comprehension with a single DB query
-        n_sessions = len([session for isub in self.get_subjects()
-                          for session in self.get_sessions(subject=isub)])
-        n_runs = len([run for isub in self.get_subjects()
-                      for run in self.get_runs(subject=isub)])
-        n_subjects = len(self.get_subjects())
+        n_subjects = len(
+            [s.value
+             for s in self.session.query(Tag).filter_by(
+                 entity_name='subject').group_by(Tag._value)]
+            )
+
+        n_sessions = len(
+            set(
+                [(t.value, t.file.entities.get('subject'))
```

Comment:
Just construct the set directly:

```suggestion
                (t.value, t.file.entities.get('subject'))
```

The overhead is trivial compared to everything else going on, but it feels cleaner not to spawn useless objects.